### ISSUE TITLE
Email not being sent because of commontator.email.undisclosed_recipients

### ISSUE THREAD

**Issue by mulander** [lml/commontator#122]
Hi,

I am using commontator with sidekiq and fastmail for sending emails.

in `app/mailers/commontator/subscriptions_mailer.rb` we have the following logic:

```ruby
  1       if uses_mailgun?                                                                                                                                                                                                        
  2         @to = recipient_emails(recipients)                                                                                                                                                                                    
  3       else                                                                                                                                                                                                                    
  4         @to = t('commontator.email.undisclosed_recipients')                                                                                                                                                                   
  5         @bcc = recipient_emails(recipients)                                                                                                                                                                                   
  6       end
```

which causes the `to:` field of the email being set to `"Undisclosed Recipients"`
emails with the `to` field set  to this value are not properly delivered even when
the `bcc` field does contain valid email addresses. I am unsure if `sidekiq` itself or `fastmail` are the ones refusing to send.

That said, it was a difficult issue to track down as the `sidekiq` jobs execute without errors. Additionally [rfc 2822](https://tools.ietf.org/html/rfc2822) defines that the `to` field if present must contain a comma separated list of addresses (3.6.3. Destination address fields) and those are also specified in the same document (3.4. Address Specification; 3.4.1. Addr-spec specification) . The string `'Undisclosed Recipients'` is not a valid address and hence due to it's presence the message could be discarded anywhere along the delivery chain resulting in undelivered email.

Some email clients display the 'Undisclosed Recipients' lines when a `to` field is not present and there is no need to add it in the header field.

Hence I suggest dropping setting the `@to` field in `subscriptions_mailer.rb` and the `commontator.email.undisclosed_recipients` translation from the plug-in as they serve no purpose.

For anyone hitting the same issue, you can workaround it by setting the translation in your `config/locales/commontator.en.yml` to an empty string which will cause the `to:` field to be unset and emails properly delivered (example below).

```yaml
en:
    commontator:
      email:
        undisclosed_recipients: ""
```

**Comment by mulander** [comment]
Swapping my config to the `rails5` branch as we speak - will let you know ASAP. Granted I am not yet using the gem in full scope but will test the features I have enabled so far.

**Comment by mulander** [comment]
Tested the branch:

 - mentions work (including sending out the email)
 - subscribing/unsubscribing to a thread works
 - commenting works
 - gravatar images work
 - I removed my config file and used the rake task to copy the new one, after
   merging my configuration everything works as expected
 - showing/hiding threads etc works
 - deleting comments work
 - this issue is fixed as without my locale workaround the emails get sent.

All in all if this branch is released as 5.0.0 then I am switching to it.

**Comment by Dantemss** [comment]
Released 5.0.0. Thank you very much.

**Comment by mulander** [comment]
Thanks for the release!

I would like to take the chance to say thank you for working on this gem. It's very reassuring to not only find a quality code base but to also find it actively maintained.

kudos.
